sdk/java/dagger-java-sdk/src/main/java/io/dagger/client/Convert.java

```diff
@@ -0,0 +1,57 @@
+package io.dagger.client;
+
+import com.google.gson.Gson;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.ExecutionException;
+
+public interface Convert {
+  static JSON toJSON(Object object)
+      throws ExecutionException, DaggerQueryException, InterruptedException {
+    Gson gson = new Gson();
+    String json;
+    if (object instanceof Scalar<?>) {
+      json = gson.toJson((((Scalar<?>) object).convert()));
+    } else if (object instanceof IDAble<?>) {
+      var id = ((IDAble<?>) object).id();
+      var idStr = ((Scalar<?>) id).convert();
+      json = gson.toJson(idStr);
+    } else {
+      json = gson.toJson(object);
+    }
+    return JSON.from(json);
+  }
+
+  static <T> T fromJSON(Client dag, JSON json, Class<T> clazz)
+      throws ClassNotFoundException,
+          InvocationTargetException,
+          NoSuchMethodException,
+          InstantiationException,
+          IllegalAccessException {
+    return fromJSON(dag, json.convert(), clazz);
+  }
+
+  static <T> T fromJSON(Client dag, String json, Class<T> clazz)
+      throws NoSuchMethodException,
+          InvocationTargetException,
+          InstantiationException,
+          IllegalAccessException,
+          ClassNotFoundException {
+    Gson gson = new Gson();
+    var o = clazz.getDeclaredConstructor().newInstance();
```

Comment:
If the type does not have a default constructor this line will throw an exception. it is better to use `isAssignableFrom()` : `Scalar.class.isAssignableFrom(clazz)` and `IDAble.class.isAssignableFrom(clazz)`